definition_validator/validator.cc

```diff
@@ -283,6 +319,8 @@ class ValidateWalk {
         validateTStructNotGrandparent(ctx.state, sym);
         validateAbstract(ctx.state, sym);
         validateAbstract(ctx.state, singleton);
+        validateFinal(ctx.state, sym, std::nullopt);
```

Comment:
SymbolRef's are optional by design. optional<SymbolRef> is redundant with `SymbolRef`. `core::Symbols::noSymbol()` is the optional value, and you can check whether a symbol is this symbol with `.exists()`